numpy/f2py/src/fortranobject.c

```diff
@@ -839,9 +839,9 @@ PyArrayObject* array_from_pyobj(const int type_num,
     if ((intent & F2PY_INTENT_INOUT) ||
             (intent & F2PY_INTENT_INPLACE) ||
             (intent & F2PY_INTENT_CACHE)) {
-        PyErr_SetString(PyExc_TypeError,
-                        "failed to initialize intent(inout|inplace|cache) "
-                        "array, input not an array");
+        PyErr_Format(PyExc_TypeError,
+                     "failed to initialize intent(inout|inplace|cache) "
+                     "array, input '%s' object is not an array", Py_TYPE(obj)->tp_name);
```

Comment:
Nit: according to the [NumPy C Style Guide](https://github.com/numpy/numpy/blob/master/doc/C_STYLE_GUIDE.rst.txt), lines should not be more than 79 characters long.

Reply:
This file is full of violations, but will fix if this needs another revision to pass CI.

Reply:
Yes, but shouldn't we strive to not make it worse?

Reply:
It did, so fixed :)

Reply:
@WarrenWeckesser: You could argue that a ~.5% increase in style quality in this file is not worth the power consumption of a full CI run. However, given CI failed that became irrelevant.